angr/analyses/decompiler/structured_codegen/c.py

```diff
@@ -914,12 +916,22 @@ def c_repr_chunks(self, indent=0, asexpr=False):
                 yield indent_str, None
             else:
                 yield " ", None
-            yield "{", brace
-            yield "\n", None
-            if node is not None:
-                yield from node.c_repr_chunks(indent=indent + INDENT_DELTA)
-            yield indent_str, None
-            yield "}", brace
+
+            omit_if_brace = (isinstance(node, CStatements) and len(node.statements) == 1) or isinstance(node, CContinue)
```

Comment:
This check is not complete. Instead of checking `isinstance(node, CContinue)`, you should check for the generic use of a single CStatement. This would be `isinstance(node, CStatement)`, which is true for CContinue.